commit ab1dd7d54840782cf24464996a8ef886adaa4659
Author: Paul Eggert <eggert@cs.ucla.edu>
Date:   Thu Jan 10 13:24:13 2013 -0800

    locale: port to Solaris 2.6 and 7 + GNU gettext
    
    * lib/locale.in.h: Just include_next <locale.h> when
    being invoked recursively.  This prevents problems on Solaris 2.6 and 7
    when combining the localename module with GNU gettext 0.18.2.
    Problem reported by Tom G. Christensen in
    <http://lists.gnu.org/archive/html/bug-gnulib/2013-01/msg00084.html>.


Above is the original commit message from gnulib.
The patch below is a rediff against the two gnulib copies in gettext 0.18.2

--- gettext-0.18.2/gettext-runtime/gnulib-lib/locale.in.h.orig	2012-12-25 06:29:09.000000000 +0100
+++ gettext-0.18.2/gettext-runtime/gnulib-lib/locale.in.h	2013-03-25 20:58:33.778086000 +0100
@@ -14,16 +14,30 @@
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 
-#ifndef _@GUARD_PREFIX@_LOCALE_H
-
 #if __GNUC__ >= 3
 @PRAGMA_SYSTEM_HEADER@
 #endif
 @PRAGMA_COLUMNS@
 
+#ifdef _GL_ALREADY_INCLUDING_LOCALE_H
+
+/* Special invocation conventions to handle Solaris header files
+   (through Solaris 10) when combined with gettext's libintl.h.  */
+
+#@INCLUDE_NEXT@ @NEXT_LOCALE_H@
+
+#else
+/* Normal invocation convention.  */
+
+#ifndef _@GUARD_PREFIX@_LOCALE_H
+
+#define _GL_ALREADY_INCLUDING_LOCALE_H
+
 /* The include_next requires a split double-inclusion guard.  */
 #@INCLUDE_NEXT@ @NEXT_LOCALE_H@
 
+#undef _GL_ALREADY_INCLUDING_LOCALE_H
+
 #ifndef _@GUARD_PREFIX@_LOCALE_H
 #define _@GUARD_PREFIX@_LOCALE_H
 
@@ -198,4 +212,5 @@
 #endif
 
 #endif /* _@GUARD_PREFIX@_LOCALE_H */
+#endif /* ! _GL_ALREADY_INCLUDING_LOCALE_H */
 #endif /* _@GUARD_PREFIX@_LOCALE_H */
--- gettext-0.18.2/gettext-tools/gnulib-lib/locale.in.h.orig	2012-12-25 06:29:42.000000000 +0100
+++ gettext-0.18.2/gettext-tools/gnulib-lib/locale.in.h	2013-03-25 20:58:50.038132000 +0100
@@ -14,16 +14,30 @@
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 
-#ifndef _@GUARD_PREFIX@_LOCALE_H
-
 #if __GNUC__ >= 3
 @PRAGMA_SYSTEM_HEADER@
 #endif
 @PRAGMA_COLUMNS@
 
+#ifdef _GL_ALREADY_INCLUDING_LOCALE_H
+
+/* Special invocation conventions to handle Solaris header files
+   (through Solaris 10) when combined with gettext's libintl.h.  */
+
+#@INCLUDE_NEXT@ @NEXT_LOCALE_H@
+
+#else
+/* Normal invocation convention.  */
+
+#ifndef _@GUARD_PREFIX@_LOCALE_H
+
+#define _GL_ALREADY_INCLUDING_LOCALE_H
+
 /* The include_next requires a split double-inclusion guard.  */
 #@INCLUDE_NEXT@ @NEXT_LOCALE_H@
 
+#undef _GL_ALREADY_INCLUDING_LOCALE_H
+
 #ifndef _@GUARD_PREFIX@_LOCALE_H
 #define _@GUARD_PREFIX@_LOCALE_H
 
@@ -198,4 +212,5 @@
 #endif
 
 #endif /* _@GUARD_PREFIX@_LOCALE_H */
+#endif /* ! _GL_ALREADY_INCLUDING_LOCALE_H */
 #endif /* _@GUARD_PREFIX@_LOCALE_H */
